controller/src/beerocks/master/db/db.cpp

```diff
@@ -2289,6 +2290,24 @@ int db::get_channel_scan_dwell_time_msec(const sMacAddr &mac, bool single_scan)
         .dwell_time_msec;
 }
 
+bool db::is_channel_scan_pool_supported(const sMacAddr &mac,
+                                        const std::unordered_set<uint8_t> &channel_pool)
+{
+    auto supported_channels = get_hostap_supported_channels(network_utils::mac_to_string(mac));
+    for (auto channel : channel_pool) {
```

Comment:
```suggestion
    for (const auto &channel : channel_pool) {
```

As far as I understand, you are only reading the `channel` value and you have no attempt to modify it, so it preferred to make it a read-only value.

Reply:
Changed